actionpack/lib/action_controller/metal/renderers.rb

```diff
@@ -16,13 +16,65 @@ def initialize(format)
       super "No renderer defined for format: #{format}"
     end
   end
+  # RFC: Consider documenting or testing MissingRenderer
+  #
+  # See <tt>Responder#api_behavior</tt>
+  # https://github.com/plataformatec/responders/blob/f0e093d01e799d8b67952c013daf8808e9ea02db/lib/action_controller/responder.rb#L209
+  #
+  #   # This is the common behavior for formats associated with APIs, such as :xml and :json.
+  #   def api_behavior
+  #     raise MissingRenderer.new(format) unless has_renderer?
+  #   end
+  #
+  #   # Check whether the necessary Renderer is available
+  #   def has_renderer?
+  #     Renderers::RENDERERS.include?(format)
+  #   end
+
+  # See <tt>Renderers.add_serializer</tt>
+  def self.add_serializer(key, &block)
+    Renderers.add_serializer(key, &block)
+  end
+
+  # See <tt>Renderers.remove_serializer</tt>
+  def self.remove_serializer(key)
+    Renderers.remove_serializer(key)
+  end
+
+  # See <tt>Renderers::MissingRenderer</tt>
+  class MissingSerializer < LoadError
+    def initialize(format)
+      super "No serializer defined for format: #{format}"
+    end
+  end
 
   module Renderers
     extend ActiveSupport::Concern
 
+    # A Set containing renderer names that correspond to available renderer procs.
+    # Default values are <tt>:json</tt>, <tt>:js</tt>, <tt>:xml</tt>.
+    RENDERERS = Set.new
```

Comment:
Here's a simple proposal for cutting down on the specialness of the configuration as much as possible:

```ruby
class ApplicationController < ActionController::Base
  serializing json: ->(object, options) { ActiveModel::SerializableResource.new(model, options) },
               xml: ->(object, options) { XmlSerializer.new(object, options) }
end

class PostsController < ApplicationController
  def show
    @post = Post.find(params[:id])
    
    respond_to do |format|
      format.json { render json: @post }
      format.xml  { render  xml: @post }
      format.csv  { render  csv: CsvSerializer.serialize(@post) }
    end
  end
end
```

Then we don't need any external registry idea. You can simply overwrite the serializing function on a per-controller basis, or in ApplicationController, if you want to set a default. This would then also mean that we constrain serializing overwrites to ApplicationController an up, which would mean this wouldn't fuck with an engine might expect.